lib/getPostcssResult.js

```diff
@@ -60,7 +60,10 @@ module.exports = function (stylelint, options = {}) {
 
 					syntax = syntaxes[stylelint._options.syntax];
 				}
-			} else if (!(options.codeProcessors && options.codeProcessors.length)) {
+			} else if (
+				!(options.codeProcessors && options.codeProcessors.length) ||
+				(options.filePath && /\.(scss|sass|less)$/.test(options.filePath))
```

Comment:
I believe it defeats the purpose of `postcss-syntax`, and removes support for many supported extensions, e. g. `.js`, `.ts`, `.html`, `.vue`.

Reply:
I'm not sure I understand. I'm only adding an `||` branch so if a file's syntax was previously inferred, then it should continue being inferred. This infers the syntax for *more* filetypes. Are you suggesting I add those extensions to this list?

For context, this is a copy-paste of the code as it was before it was accidentally removed in https://github.com/stylelint/stylelint/pull/4729/files#diff-5383a0a864da5a5ad24b4bae8f44c5574601bd4d145999b1ddc211d3900dd68bL89

Thanks for taking a look :)

Reply:
> I understand if you still don't want to merge this but i've spent a couple hours on this

@yaymukund Thank you for taking the time to dig into this. The syntax hullabaloo in stylelint is gnarly. I'm personally looking forward to the possibility of removing it in favour of explicit-configuration using a `overrides` property. https://github.com/stylelint/stylelint/pull/5126 is the first step towards that.

Anyhow, in the meantime, would you mind removing the `else if` in favour of simple `else`, i.e. autosyntax is always used regardless of if a processor is configured in the configuration object?:

```
} else { 
	const autoSyntax = require('postcss-syntax');

	// TODO: investigate why lazy import HTML syntax causes
	// JS files with the word "html" to throw TypeError
	// https://github.com/stylelint/stylelint/issues/4793
	const { html, ...rest } = syntaxes;

	syntax = autoSyntax({
		css: cssSyntax(stylelint),
		jsx: syntaxes['css-in-js'],
		...rest,
	});
}
```

And seeing if that works in your repository.

> (We actually ended up passing -s scss to work around this issue, so it's no longer blocking us.)

Ha, yes, that'll do it :)

Reply:
It looks like that condition has existed for as long as postcss-safe-parser:

https://github.com/stylelint/stylelint/pull/3261/files#diff-5383a0a864da5a5ad24b4bae8f44c5574601bd4d145999b1ddc211d3900dd68bR89

But changing it to `else` seems to work for our repo, and the tests here pass, so this should be GTM if all that checks out for you :)

Re: syntax configuration: that makes sense. I would be in favor of an explicit opt-in API, if only so we can more easily reason about what's happening.

Reply:
Thanks for making the change and testing it with your repo.

@hudochenkov what do you think, can we merge this or would you rather hold off while you're looking into [what the future of syntaxes could be](https://github.com/postcss/postcss/issues/1498)?